### PR TITLE
Add Kokoro CI job for publishing snapshots to Maven Central

### PR DESCRIPTION
**Pull request by meltsufin**
Adds `continous.cfg` build configuration for Kokoro. The job first builds the project skipping integration tests, and then deploys (publishes) the snapshot version to Sonatype Snapshots Repository. See: https://oss.sonatype.org/content/repositories/snapshots/com/google/cloud/cloud-spanner-r2dbc/.

This PR also adds all of the [required](https://central.sonatype.org/pages/requirements.html) metadata for publishing to Maven Central.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -9,6 +9,71 @@
...
+      <id>dmitry-s</id>
+      <name>Dmitry Solomakha</name>
+      <email>dsolomakha@google.com</email>
+      <organization>Google</organization>
+      <organizationUrl>http://www.google.com</organizationUrl>
+    </developer>
+    <developer>
+      <id>dzou</id>
+      <name>Daniel Zou</name>
+      <email>dzou@google.com</email>
```
Comment (dzou) [inline pom.xml]: Do we have to include our emails here? Feel like we will get spammed. :sweat_smile:

Comment (meltsufin) [reply pom.xml]: You already include your email in the commit. Don't worry, we have good spam filters. :D

Comment (elefeint) [reply pom.xml]: "Come to the dark side; we have cookies!"

**Comment by meltsufin** [comment]
Regarding the metadata requirements, see: https://central.sonatype.org/pages/requirements.html

**Comment by elefeint** [comment]
"It is acceptable to link to your profile on GitHub or other forges, if you do no have a website." -- so can we use the `@users.noreply.github.com` e-mail addresses?

**Pull request merged**